### PR TITLE
feat: added legal notice component for imprint and privacy

### LINKED ISSUE #8427: add legal notice for imprint and privacy
### Is your proposal related to a problem?

We need legal notices for the public pages. There should be a possibility to link an imprint and a privacy policy. The URLs are configurable in .env file

grafik
grafik
grafik

### PR DESCRIPTION
**Pull request by servusrene**
## What does this PR do?

fixes #8427

### CODE REVIEW

File: apps/web/components/ui/LegalNotice.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    return null;
+  }
+
+  return (
+    <div className="p-2 text-center text-xs sm:text-right">
+      {IMPRINT_URL ? (
+        <a
+          target="_blank"
+          href={IMPRINT_URL}
+          className="opacity-50 hover:opacity-100 dark:text-white"
```
Comment (sean-brydon) [inline apps/web/components/ui/LegalNotice.tsx:19]:
```suggestion
          className="opacity-50 hover:opacity-100 text-emphasis"
```

File: apps/web/components/ui/LegalNotice.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          rel="noreferrer">
+          {t("Imprint")}
+        </a>
+      ) : null}
+      {IMPRINT_URL && PRIVACY_URL ? <span className="opacity-50 dark:text-white">{" | "}</span> : null}
```
Comment (sean-brydon) [inline apps/web/components/ui/LegalNotice.tsx:24]:
```suggestion
      {IMPRINT_URL && PRIVACY_URL ? <span className="opacity-50 text-emphasis">{" | "}</span> : null}
```

File: apps/web/components/ui/LegalNotice.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      {PRIVACY_URL ? (
+        <a
+          target="_blank"
+          href={PRIVACY_URL}
+          className="opacity-50 hover:opacity-100 dark:text-white"
```
Comment (sean-brydon) [inline apps/web/components/ui/LegalNotice.tsx:29]:
```suggestion
          className="opacity-50 hover:opacity-100 text-emphasis"
```

**Review by sean-brydon — CHANGES_REQUESTED** [review]
Recently we added tokens to cover dark+light mode in one class!

Would be great if we can use them here too :)

**Review by sean-brydon — CHANGES_REQUESTED** [review]
LGTM thanks for making them changes also 

Would it be possible to remove yarn.lock from this PR

`git checkout origin/main -- yarn.lock` 

Should do the trick :)

**Comment by sean-brydon** [comment]
LGTM 

@PeerRich  @Jaibles is this something we should possibly hide when the user has branding disabled?

**Comment by sean-brydon** [comment]
> @sean-brydon as these are legal requirements, that would not be good.

Fair point @Jaibles is there somewhere better we can put these? I know a lot of users like the clean luck of nothing below the  booker

File: apps/web/components/ui/LegalNotice.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+const LegalNotice = () => {
+  const { t } = useLocale();
+  const isEmbed = useIsEmbed();
+
+  if (isEmbed) {
+    return null;
+  }
+
+  return (
+    <div className="p-2 text-center text-xs sm:text-right">
```
Comment (PeerRich) [inline apps/web/components/ui/LegalNotice.tsx:14]: this empty container would render if there is no variable set

File: apps/web/pages/[user].tsx [hunk 5]
```diff
@@ -33,6 +33,7 @@ import type { inferSSRProps } from "@lib/types/inferSSRProps";
 import type { EmbedProps } from "@lib/withEmbedSsr";
 
 import PageWrapper from "@components/PageWrapper";
+import LegalNotice from "@components/ui/LegalNotice";
```
Comment (PeerRich) [inline apps/web/pages/[user].tsx:36]: @sean-brydon this should be in features not UI components no?

**Review by PeerRich — CHANGES_REQUESTED** [review]

File: apps/web/pages/[user].tsx (hunk 5 above)
Comment (rene-mueller) [reply apps/web/pages/[user].tsx:36]: @PeerRich i have it there because PoweredByCal is also located there.

Comment (PeerRich) [reply apps/web/pages/[user].tsx:36]: ok, talking to sean now where to put it. we can merge afterwards

File: apps/web/components/ui/LegalNotice.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      {IMPRINT_URL ? (
+        <a
+          target="_blank"
+          href={IMPRINT_URL}
+          className="text-emphasis opacity-50 hover:opacity-100"
+          rel="noreferrer">
+          {t("Imprint")}
```
Comment (PeerRich) [inline apps/web/components/ui/LegalNotice.tsx:25]:
```suggestion
          {t("imprint")}
```

**Review by PeerRich — COMMENTED** [review]
both "privacy" and "imprint" need to be added to /en/common.json

**Comment by rene-mueller** [comment]
@PeerRich are there any updates?

**Review by leog — CHANGES_REQUESTED** [review]
Thank you for your contribution @rene-mueller. It would be best to avoid adding more env vars and have the legal notice variables as part of the database, probably in the `Deployment` table where information about the deployment is stored. Would you be able to tackle this?

**Comment by rene-mueller** [comment]
@leog sure i can move it to the `Deployment ` table in the database. I haven't worked with prisma yet. Is there a guide to create migration files available?

**Comment by rene-mueller** [comment]
@leog what do you think about: #9072

**Comment by PeerRich** [comment]
> @leog what do you think about: #9072

https://github.com/calcom/cal.com/issues/9072#issuecomment-1562584912

**Comment by PeerRich** [comment]
hey @rene-mueller the reason we havent merged this, is because we were in the middle of launching our entire new booking page.

we are very open now to add imprint and privacy to the booking page, but it would need to be via database variables and not via .env variables. would you be open to raise a new PR with this for the new booking page?


how do you like this?

image

**Pull request closed without merging**